design-documents/graph-ql/coverage/returns.md

```diff
@@ -0,0 +1,350 @@
+{
+  pageSpecificCustomAttributes(page_type: RETURN_ITEM_EDIT_FORM) {
+    items {
+      id
+      attribute_code
+      attribute_type
+      input_type
+      attribute_options {
+        id
+        label
+        value
+      }
+    }
+  }
+}
+```
+
+Existing schema of `Attribute` and `AttributeOption` must be extended to provide `ID` field, which will be used in mutations to specify custom attribute values for returns.
+
+#### Determine which order items are eligible for return
+
+```graphql
+{
+  customer {
+    orders(filter: {number: {eq: "00000008"}}) {
+      items {
+        items_eligible_for_return {
+          id
+          product_name
+        }
+      }
+    }
+  }
+}
+```
+
+#### Submit a return with multiple items and comments
+
+```graphql
+mutation {
+  createReturn(
+    input: {
+      items: [
+        {
+          order_item_id: "0000004", 
+          quantity_to_return: 1, 
+          selected_custom_attributes: ["encoded-custom-select-attribute-value-id"], 
+          entered_custom_attributes: [{id: "encoded-custom-text-attribute-id", value: "Custom attribute value"}]
+        }
+      ], 
+      comment_text: "Return comment"
+    }
+  ) {
+    return {
+      id
+      items {
+        id
+        quantity
+        request_quantity
+        product {
+          sku
+          name
+        }
+        custom_attributes {
+          id
+          label
+          value
+        }
+      }
+      comments {
+        created_at
+        created_by
+        text
+      }
+    }
+  }
+}
+```
+
+Alternatively, the collection of returns associated with the order can be requested.
+
+### Leave a return comment
+
+```graphql
+mutation {
+  addReturnComment(
+    input: {
+      return_id: "000000001", 
+      comment_text: "Another return comment"
+    }
+  ) {
+    return {
+      id
+      comments {
+        created_at
+        created_by
+        text
+      }
+    }
+  }
+}
+```
+
+### Create a return for guest order
+
+Guest orders are not accessible via GraphQL yet, but the schema of returns will be identical to the one for customer orders.
+
+### Specify shipping and tracking
+
+When return is authorized by the admin user, the customer can specify shipping and tracking information.
+
+First, the client needs to get shipping cariers that can be used for returns:
+
+```graphql
+{
+  customer {
+    return(id: "0000003") {
+      available_shipping_carriers {
+        id
+        label
+      }
+    }
+  }
+}
+```
+
+Then tracking information can be submitted:
+
+```graphql
+mutation {
+  addReturnTracking(
+    input: {
+      return_id: "000005", 
+      carrier_id: "carrier-id", 
+      tracking_number: "4234213"
+    }
+  ) {
+    return {
+      shipping {
+        tracking {
+          id
+          carier {
+            label
+          }
+          tracking_number
+        }
+      }
+    }
+  }
+}
+```
+
+If the user decides to view the status of the return, it can be retrieved using the following query:
```

Comment:
> If the user decides to view the status of the return, it can be retrieved using the following query

Think this query is for the status of the tracking on the return for a specific tracking item, right?

Reply:
Changed to `If the user decides to view the status of the specific tracking item, it can be retrieved using the following query`